crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -76,6 +68,34 @@ pub(crate) struct ObjectKey {
     version: u64,
 }
 
+#[derive(Union, Clone)]
+pub enum ObjectOwnerNew {
+    Immutable(Immutable),
+    Shared(Shared),
+    Parent(Parent),
+    Address(AddressOwner),
+}
+
+#[derive(SimpleObject, Clone)]
+pub struct Immutable {
+    dummy: Option<bool>,
```

Comment:
The schema looks like this so not sure if this is the right way
```
type Immutable {
  # Dummy field
  _: Boolean
}
```

Reply:
I think this is because we can't have scalar types in a union type. I'm not sure if the syntax in the draft schema will work though (not sure how we can get the _), so resolving would be ... on Immutable { dummy }? @amnn or @hayes-mysten please correct me if I'm wrong

Reply:
I think @wlmyng found the trick I used for this in `programmable.rs`, but I don't see the reply on this inline comment, so replicating it here:

https://github.com/MystenLabs/sui/blob/9ba167b0af84365188877a448a733a5d877e4ba5/crates/sui-graphql-rpc/src/types/transaction_block_kind/programmable.rs#L167-L174

This at least was the intention, but if there is some neater way, I'll take it.  The issue is not about having scalars in union types (I'm not sure that's an issue?) but that if you have an empty type in your GraphQL schema, GraphiQL's doc generator really doesn't like it.

Reply:
Could we also get ahead of the need to document things by adding docs to these new types?

Reply:
Added